src/accordion/accordion.directive.ts

```diff
@@ -0,0 +1,305 @@
+			if (!this._collapsed) {
+				this.animatingBodyCollapse = true;
+				this.cd.detectChanges();
+			}
+			// we also need to make sure 'animation' flag is up-to- date
+			this.collapse.ngbCollapse.animation = this.accordion.animation;
+			this.collapse.ngbCollapse.collapsed = collapsed;
+		}
+	}
+
+	@Output() shown = new EventEmitter<void>();
+	@Output() hidden = new EventEmitter<void>();
+
+	get collapsed(): boolean {
+		return this._collapsed;
+	}
+	get disabled() {
+		return this._disabled;
+	}
+	public get id() {
+		return `ngb-accordion-item-${this._id}`;
+	}
+
+	get toggleId() {
+		return `ngb-accordion-toggle-${this._id}`;
+	}
+	get collapseId() {
+		return `ngb-accordion-collapse-${this._id}`;
+	}
+
+	ngAfterContentInit(): void {
+		const { ngbCollapse } = this.collapse;
+		// we need to disable the animation for the first init
+		ngbCollapse.animation = false;
+		ngbCollapse.collapsed = this.collapsed;
+		// we set the animation to the default of the accordion
+		ngbCollapse.animation = this.accordion.animation;
+		// event forwarding from 'ngbCollapse' to 'ngbAccordion'
+		this.subscriptions.push(
+			ngbCollapse.hidden.subscribe(() => {
+				// when the animation finishes we can remove the template from DOM
+				this.animatingBodyCollapse = false;
+				this.hidden.emit();
+				this.accordion.hidden.emit(this.id);
+			}),
+			ngbCollapse.shown.subscribe(() => {
+				this.shown.emit();
+				this.accordion.shown.emit(this.id);
+			}),
+		);
+	}
+
+	toggle() {
+		this.collapsed = !this.collapsed;
+	}
+}
+
+@Directive({
+	exportAs: 'ngbAccordion',
+	standalone: true,
+	selector: '[ngbAccordion]',
+	host: { '[class.accordion]': 'true' },
+})
+export class NgbAccordionDirective {
+	@ContentChildren(NgbAccordionItem, { descendants: false }) private items: QueryList<NgbAccordionItem>;
+	/**
+	 * If `true`, accordion will be animated.
+	 */
+	@Input() animation: boolean;
+	/**
+	 * If `true`, only one item at the time can stay open.
+	 */
+	@Input() closeOthers: boolean;
+	/**
+	 * If `true`, the content of the AccordionBody will be removed from the DOM.
+	 * This property can be overwritten at AccordionItem level
+	 */
+	@Input() destroyOnHide = true;
+
+	/**
+	 * An event emitted when the expanding animation is finished on the collapse. The payload is the AccordionItem id.
+	 */
+	@Output() shown = new EventEmitter<string>();
+
+	/**
+	 * An event emitted when the collapsing animation is finished on the collapse, and before the content of the AccordionBody is removed.
+	 * The payload is the AccordionItem id.
+	 */
+	@Output() hidden = new EventEmitter<string>();
+
+	constructor(config: NgbAccordionConfig) {
+		this.animation = config.animation;
+		this.closeOthers = config.closeOthers;
+	}
+
+	/**
+	 * Toggles a panel with the given id.
+	 *
+	 * Has no effect if the panel is disabled.
+	 */
+	toggle(itemId: string) {
+		if (this.canBeToggled(itemId)) {
```

Comment:
doing `this.canBeToggled` seemed like a lot of redundant work, but indeed I don't see how to do it better...